v1/src/lib.rs

```diff
@@ -299,11 +318,106 @@ impl Volume for InfluxDbVolume {
+                                batch_start_time = Instant::now();
+                            }
+                            None => {
+                                debug!("[{}] batch put channel closed, exiting task", name_clone,);
+                                break;
+                            }
+                        }
+                    } else {
+                        // ...and wait for more items
+                        match tokio::time::timeout(Duration::from_millis(100), rx.recv()).await {
+                            Ok(r) => match r {
+                                Some(Put { query, measurement }) => {
+                                    // add to batch
+                                    put_batch.push(query);
+                                    measurement_counts
+                                        .entry(measurement)
+                                        .and_modify(|counter| *counter += 1)
+                                        .or_insert(1);
+                                }
+                                None => {
+                                    debug!(
+                                        "[{}] batch put channel closed, exiting task",
+                                        name_clone
+                                    );
+                                    break;
+                                }
+                            },
+                            Err(_) => {
+                                // timeout
+                            }
+                        };
+
+                        let batch_full = put_batch.len() >= put_batch_size;
+                        let batch_timed_out = batch_start_time.elapsed() > put_batch_timeout;
+
+                        if batch_full || batch_timed_out {
+                            let n = put_batch.len();
+
+                            let counts: Vec<String> = measurement_counts
+                                .drain()
+                                .map(|(k, v)| format!("{k} x {v}"))
+                                .collect();
+                            debug!(
+                                "[{}] PUT batch of {} - {}",
+                                name_clone,
+                                n,
+                                counts.join(", ")
+                            );
+                            let result = client_clone.query(&put_batch).await;
+                            put_batch.clear();
+
+                            if let Err(e) = result {
+                                debug!(
+                                    "[{}] Failed to put Value for batch of {} in InfluxDb storage : {}", name_clone,
+                                    n, e
+                                )
+                            }
+                        }
+                    }
+                }
+            });
+
+            Some(tx)
+        } else {
+            None
+        };
```

Comment:
Suggestion / nit: `let mut put_batch_tx = None;` above 322 and just over write with a sum value it internally.